### PR TITLE
subsys: bluetooth: hids: Header with common data structures

### PR DESCRIPTION
**Pull request by KAGA164**
This commit adds a common header for hids services with
common data structures.

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

### CODE REVIEW

**Comment by KAGA164** [comment]
@rakons @kapi-no please review

File: applications/nrf_desktop/src/modules/hid_forward.c [hunk 1]
```diff
@@ -194,7 +194,7 @@ static void hidc_ready(struct bt_gatt_hids_c *hids_c)
 
 	while (NULL != (rep = bt_gatt_hids_c_rep_next(hids_c, rep))) {
 		if (bt_gatt_hids_c_rep_type(rep) ==
-		    BT_GATT_HIDS_C_REPORT_TYPE_INPUT) {
+		    BT_GATT_HIDS_COMMON_REPORT_TYPE_INPUT) {
```
Comment (rakons) [inline applications/nrf_desktop/src/modules/hid_forward.c]: Use rather BT_GATT_HIDS_REPORT_TYPE_INPUT. Remove the COMMON word - it provides no information only making it harder to understand.

**Comment by carlescufi** [comment]
@joerchan please review

File: include/bluetooth/services/hids_common.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
+/*
```
Comment (joerchan) [inline include/bluetooth/services/hids_common.h]: Can't this be added to hids.h instead of creating a new header file?

Comment (KAGA164) [reply include/bluetooth/services/hids_common.h]: Ok. I moved all common data structures to hids.h header and include it to hid_c.h. Probablly this is more readable now.

Comment (rakons) [reply include/bluetooth/services/hids_common.h]: Is hids_common.h file still here or am I missing something?

**Pull request merged**